slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -329,6 +325,110 @@ void SLAMServiceImpl::SetUpMapBuilder() {
 }
 
 std::string SLAMServiceImpl::GetLatestJpegMapString(bool add_pose_marker) {
+    
+    std::unique_ptr<cartographer::io::PaintSubmapSlicesResult> painted_slices = nullptr;
+    try {
+        painted_slices = std::make_unique<cartographer::io::PaintSubmapSlicesResult>(GetLatestPaintedMapSlices());
+    } catch (std::exception &e) {
+        // jpeg will be empty which captures the error
```

Comment:
[minor] can we log its an empty jpeg?

Reply:
We capture this log in the grpc server [here](https://github.com/viamrobotics/slam/pull/136/files#diff-f582be4089c313a31e0dfa10d48874bfa84a0fc08a2983d6ceea52b161e574d6R136), can add a `LOG` line if we think we need more

Reply:
[nit] `// jpeg will be empty which captures the error` might be better as `// capture error from painting submaps if no submaps are present`. Or could remove and just have the log